src/libertem/io/dataset/base/dataset.py

```diff
@@ -81,6 +86,8 @@ def get_num_partitions(self) -> int:
         native dtype. At least :code:`self._cores` partitions
         are created.
         """
+        if self._user_num_partitions is not None:
+            return self._user_num_partitions
```

Comment:
Since there is no situation where zero or negative partitions could make sense, better to clip this here early rather than get a strange index out of range or div-by-0 error later? Could also `raise` explicitly of course.

```suggestion
            return max(1, self._user_num_partitions)
```